janus_core/cli.py

```diff
@@ -151,21 +160,107 @@ def singlepoint(
         device=device,
         read_kwargs=read_kwargs,
         calc_kwargs=calc_kwargs,
-        log_file=log_file,
+        log_kwargs={"filename": log_file, "filemode": "w"},
     )
     s_point.run_single_point(
         properties=properties, write_results=True, write_kwargs=write_kwargs
     )
 
 
 @app.command()
-def test(name: str):
+def geomopt(  # pylint: disable=too-many-arguments,too-many-locals
+    struct_path: StructPath,
+    fmax: Annotated[
+        float, typer.Option("--max-force", help="Maximum force for convergence")
+    ] = 0.1,
+    architecture: Architecture = "mace_mp",
+    device: Device = "cpu",
+    fully_opt: Annotated[
+        bool,
+        typer.Option("--fully-opt", help="Optimise cell and atomic positions"),
+    ] = False,
+    vectors_only: Annotated[
+        bool,
+        typer.Option("--vectors-only", help="Allow only hydrostatic deformations"),
+    ] = False,
+    read_kwargs: ReadKwargs = None,
+    calc_kwargs: CalcKwargs = None,
+    write_kwargs: WriteKwargs = None,
+    traj_file: Annotated[
+        str, typer.Option("--traj", help="Path to save trajectory to")
+    ] = None,
+    log_file: LogFile = "geomopt.log",
+):
     """
-    Dummy alternative CLI command.
+    Perform geometry optimization and save to file.
```

Comment:
Save what to file? Final state? All intermediate states?

Reply:
both

Reply:
I've written it as the optimized structure, as that's the default behaviour, and hopefully it's clear from the options that both are true.